lib/elixir/lib/calendar.ex

```diff
@@ -314,6 +321,23 @@ defmodule Date do
     end
   end
 
+  @doc """
+  Computes the day of the week of a given `Date` struct.
+
+  Returns the day of the week as 1: Monday, 2: Tuesday, and so on.
+
+  ## Examples
+
+      iex> Date.day_of_week(~D[2016-10-31])
+      1
+      iex> Date.day_of_week(~D[2016-11-01])
+      2
+  """
+  @spec day_of_week(Date.t) :: integer
+  def day_of_week(%Date{calendar: calendar, year: year, month: month, day: day}) do
+    calendar.day_of_week(year, month, day)
```

Comment:
No I didn't.

Because we want to use the `day_of_week` function from the date's `calendar` which can be `ISO` or any other calendar.